File: find_code.js
const fs = require('fs');
const content = fs.readFileSync('src/components/TektonApp.tsx', 'utf8');

const regex = /appointments\.filter\([^)]*\)\.map\(/;
const match = content.match(regex);

if (match) {
    const idx = match.index;
    console.log(content.slice(Math.max(0, idx + 6500), idx + 8000));
}

File: find_code.py
import sys

with open('src/components/TektonApp.tsx', 'r', encoding='utf8') as f:
    content = f.read()

idx = content.find('appointments.filter(')
if idx == -1:
    idx = content.find('appointments.map(')

if idx != -1:
    print(content[max(0, idx - 100):idx + 1000])
else:
    print("Could not find appointments rendering logic.")
